src/app/GitCommands/Settings/GitConfigSettings.cs

```diff
@@ -0,0 +1,110 @@
+#nullable enable
+
+using System.Diagnostics;
+using GitCommands.Git;
+using GitExtensions.Extensibility;
+using GitExtensions.Extensibility.Configurations;
+using GitExtensions.Extensibility.Git;
+
+namespace GitCommands.Settings;
+
+[DebuggerDisplay("{" + nameof(DebuggerDisplay) + ",nq}")]
+public sealed class GitConfigSettings : GitConfigSettingsBase, IPersistentConfigValueStore
+{
+    private readonly Dictionary<string, string?> _modifiedSettings = [];
+    private readonly HashSet<string> _multiValueSettings = [];
+
+    public GitConfigSettings(IExecutable gitExecutable, GitSettingLevel gitSettingLevel)
+        : base(gitExecutable, gitSettingLevel)
+    {
+        if (gitSettingLevel == GitSettingLevel.Effective)
+        {
+            throw new ArgumentOutOfRangeException(nameof(gitSettingLevel), $"Use read-only {nameof(EffectiveGitConfigSettings)} instance instead.");
```

Comment:
[`ArgumentOutOfRangeException`](https://learn.microsoft.com/dotnet/api/system.argumentoutofrangeexception?view=net-9.0#methods) has a number of Throw* helpers, may worth using those.

Reply:
> [`ArgumentOutOfRangeException`](https://learn.microsoft.com/dotnet/api/system.argumentoutofrangeexception?view=net-9.0#methods) has a number of Throw* helpers, may worth using those.

already checked, not enough: it is unable to compare enum values

Reply:
These aren't flags, so compare as ints?